### PR TITLE
fix: Changes to zkstack after testing migration from GW

### PR DESCRIPTION
**Pull request by StanislavBreadless**
## What ❔

Mainly some additional logs for better understanding of upgrade status, but also some corner case fixes

## Is this a breaking change?
- [ ] Yes
- [ ] No

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs [hunk 1]
```diff
@@ -201,21 +226,16 @@ pub(crate) async fn get_gateway_migration_state(
...
         }
 
         let Some(current_sl_from_server) = gateway_migration_status.settlement_layer else {
-            todo!(
-                "
-                The migration has started, but the settlement layer is not defined. \
-                Waiting manual finalization or L2 L1 transaction"
-            );
+            // It means that the server is in the middle of some migration.
+            anyhow::bail!("Very old migration in progress");
```
Comment (Deniallugo) [inline zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs:238]: Using direction, you can return that it requires manual finalization. or waiting for gateway to process the tx

Comment (StanislavBreadless) [reply zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs:238]: This entire clause is for the case when I could not find L1 migration event on L1, so the event is either very old or there has been no migration before. 

If the event is too old, we dont handle such case yet

File: zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs [hunk 2]
```diff
@@ -201,21 +226,16 @@ pub(crate) async fn get_gateway_migration_state(
...
+        // Or the latest migration was long time ago.
 
         // Firslty check for consistency with the server.
-        if gateway_migration_status.latest_notification.is_some() {
-            anyhow::bail!(format!(
-                "Server has seen an event not present on L1. Status {:#?}",
-                gateway_migration_status
-            ));
+        if gateway_migration_status.state == GatewayMigrationState::InProgress {
+            anyhow::bail!("Very old migration in progress");
```
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs:233]: same as aboev

[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply zkstack_cli/crates/zkstack/src/commands/chain/gateway/gateway_common.rs:233]: you can read the meaning of this case in the comments above
